crates/bitwarden/src/platform/fido2/mod.rs

```diff
@@ -55,18 +74,129 @@ impl<'a> ClientFido2<'a> {
-            .ok_or("No Fido2Credential")?;
+    fn try_from(value: CipherViewContainer) -> Result<Self, Self::Error> {
+        let cred = value
+            .fido2_credentials
+            .first()
+            .ok_or(Error::Internal("No Fido2 credentials found".into()))?;
+
+        cred.clone().try_into()
+    }
+}
+
+impl TryFrom<Fido2CredentialFullView> for Passkey {
+    type Error = crate::error::Error;
+
+    fn try_from(value: Fido2CredentialFullView) -> Result<Self, Self::Error> {
+        let counter: u32 = value.counter.expose().parse().expect("Invalid counter");
+        let counter = (counter != 0).then_some(counter);
+
+        let key = pkcs8_to_cose_key(value.key_value.expose().as_ref())?;
+
+        Ok(Self {
+            key,
+            credential_id: string_to_guid_bytes(value.credential_id.expose())?.into(),
+            rp_id: value.rp_id.expose().clone(),
+            user_handle: value.user_handle.map(|u| u.expose().to_vec().into()),
+            counter,
+        })
+    }
+}
+
+impl Fido2CredentialFullView {
+    pub(crate) fn try_from_credential(
+        value: Passkey,
+        user: passkey::types::ctap2::make_credential::PublicKeyCredentialUserEntity,
+        rp: passkey::types::ctap2::make_credential::PublicKeyCredentialRpEntity,
+    ) -> Result<Self> {
+        let cred_id: Vec<u8> = value.credential_id.into();
+
+        Ok(Fido2CredentialFullView {
+            credential_id: SensitiveString::new(Box::new(guid_bytes_to_string(&cred_id)?)),
+            key_type: SensitiveString::new(Box::new("public-key".to_owned())),
+            key_algorithm: SensitiveString::new(Box::new("ECDSA".to_owned())),
+            key_curve: SensitiveString::new(Box::new("P-256".to_owned())),
+            key_value: SensitiveVec::new(Box::new(cose_key_to_pkcs8(&value.key)?)),
+            rp_id: SensitiveString::new(Box::new(value.rp_id)),
+            rp_name: rp.name.map(|n| SensitiveString::new(Box::new(n))),
+            user_handle: Some(SensitiveVec::new(Box::new(cred_id))),
+
+            counter: SensitiveString::new(Box::new(value.counter.unwrap_or(0).to_string())),
+            user_name: user.name.map(|n| SensitiveString::new(Box::new(n))),
+            user_display_name: user.display_name.map(|n| SensitiveString::new(Box::new(n))),
+            discoverable: SensitiveString::new(Box::new("true".to_owned())),
+            creation_date: chrono::offset::Utc::now(),
+        })
+    }
+}
+
+pub fn guid_bytes_to_string(source: &[u8]) -> Result<String> {
+    if source.len() != 16 {
+        return Err(Error::Internal("Input should be a 16 byte array".into()));
+    }
+    Ok(uuid::Uuid::from_bytes(source.try_into().expect("Invalid length")).to_string())
+}
+
+pub fn string_to_guid_bytes(source: &str) -> Result<Vec<u8>> {
+    let Ok(uuid) = uuid::Uuid::parse_str(source) else {
+        return Err(Error::Internal("Input should be a valid GUID".into()));
+    };
+    Ok(uuid.as_bytes().to_vec())
+}
```

Comment:
issue: I think me might need to roll our own conversion. We've had issues with UUID conversions on other platforms because I don't think we're using fully compliant UUIDs. I remember this PR https://github.com/bitwarden/clients/pull/6806 and I've asked Vault for more information https://bitwarden.slack.com/archives/C04N1TX1CGM/p1716819529648879

Reply:
While you're at it, do you think that you could add a Base64URL decoding options as well? If you think that increases the scope of the PR too much just let me know and I'll create a separate PR for it!

For more information see: [[SDK] Add support for raw byte credential id](https://bitwarden.atlassian.net/browse/PM-7380)

Reply:
Code snippet looks good! I'll give the whole PR final review just to make sure then I'll go over and approve this!